e2etest/netpoltests/networkpolicies.go

```diff
@@ -0,0 +1,161 @@
+
+		want, err := expectedPolicies()
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(len(got.Items)).To(Equal(len(want)),
+			"Want %d network policies, got %d", len(want), len(got.Items))
+
+		for _, wantPolicy := range want {
+			found := false
+			for _, gotPolicy := range got.Items {
+				if gotPolicy.Name == wantPolicy.Name && gotPolicy.Namespace == wantPolicy.Namespace {
+					diff := cmp.Diff(wantPolicy.Spec, gotPolicy.Spec)
+					Expect(diff).To(BeEmpty(), "Policy spec mismatch for %s/%s:\n%s",
+						wantPolicy.Namespace, wantPolicy.Name, diff)
+
+					found = true
+					break
+				}
+			}
+			Expect(found).To(BeTrue(), "Policy %s/%s not found", wantPolicy.Namespace, wantPolicy.Name)
+		}
+
+		probeNamespace, err := k8s.CreateTestNamespace(cs, "test-netpol")
+		Expect(err).NotTo(HaveOccurred())
+		ginkgo.DeferCleanup(func() {
+			err = k8s.DeleteNamespace(cs, probeNamespace)
+			Expect(err).NotTo(HaveOccurred())
+		})
+		tpod := &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "netpol-test",
+				Namespace: probeNamespace,
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{{
+					Name:  "netpol",
+					Image: agnhostImage,
+				}},
+			},
+		}
+		probe, err = metallb.CreatePod(cs, tpod)
+		Expect(err).NotTo(HaveOccurred())
+		ginkgo.DeferCleanup(func() {
+			err = metallb.DeletePod(cs, probe)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		controller, err = metallb.ControllerPod(cs)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	ginkgo.It("only allowed traffic", func() {
+		ginkgo.By("checking ingress - probe pod to any port on the controller other than webhook should be timeout")
+		controllerIP := controller.Status.PodIP
+		probeExec := executor.ForPod(probe.Namespace, probe.Name, "netpol")
+		out, err := probeExec.Exec("./agnhost", "connect", net.JoinHostPort(controllerIP, "7472"), "--timeout", "5s")
+		Expect(err).To(HaveOccurred())
+		Expect(out).To(ContainSubstring("TIMEOUT"), out)
+		out, err = probeExec.Exec("./agnhost", "connect", net.JoinHostPort(controllerIP, "9443"), "--timeout", "5s")
+		Expect(err).NotTo(HaveOccurred(), out) // Until we manage to allow traffic only from k8s API
+
+		ginkgo.By("checking egress - controller pod to any port on the probe other port than API should be timeout")
+		probeIP := probe.Status.PodIP
+		ctrlExec, err := executor.ForPodDebug(cs, controller.Namespace, controller.Name, "controller")
+		Expect(err).NotTo(HaveOccurred())
+		out, err = ctrlExec.Exec("./agnhost", "connect", net.JoinHostPort(probeIP, "8080"), "--timeout", "5s")
+		Expect(err).To(HaveOccurred())
+		Expect(out).To(ContainSubstring("TIMEOUT")) // If no netpol then REFUSED
+		out, err = ctrlExec.Exec("./agnhost", "connect", net.JoinHostPort(probeIP, "6443"), "--timeout", "5s")
+		Expect(err).To(HaveOccurred())
+		Expect(out).To(ContainSubstring("REFUSED")) // until we restrict only to k8s svc
+	})
+})
+
+func expectedPolicies() ([]networkingv1.NetworkPolicy, error) {
```

Comment:
I think this is unnecessary (checking that the objs are there), e.g for the same reasons we don't check the webhook objects are there before running the webhook e2es. I guess we can keep it if you see there's value in this

Reply:
Agree on this being redundant, we want to ensure the behavior, not the configuration that leads to the behavior.